pkg/cache/clusterqueue.go

```diff
@@ -165,8 +185,20 @@ var defaultPreemption = kueue.ClusterQueuePreemption{
 
 var defaultFlavorFungibility = kueue.FlavorFungibility{WhenCanBorrow: kueue.Borrow, WhenCanPreempt: kueue.TryNextFlavor}
 
-func (c *clusterQueue) update(in *kueue.ClusterQueue, resourceFlavors map[kueue.ResourceFlavorReference]*kueue.ResourceFlavor, admissionChecks map[string]AdmissionCheck) error {
-	c.updateResourceGroups(in.Spec.ResourceGroups)
+func (cache *Cache) updateClusterQueue(c *clusterQueue, in *kueue.ClusterQueue, resourceFlavors map[kueue.ResourceFlavorReference]*kueue.ResourceFlavor, admissionChecks map[string]AdmissionCheck) error {
+	oldCohort := c.Parent()
+	cache.hm.UpdateClusterQueueEdge(in.Name, in.Spec.Cohort)
+	if c.updateResourceGroups(in.Spec.ResourceGroups) || oldCohort != c.Parent() {
+		updateClusterQueueResourceNode(c)
+		c.AllocatableResourceGeneration += 1
+		if oldCohort != nil && oldCohort != c.Parent() {
+			updateCohortResourceNode(oldCohort)
+		}
+		if c.HasParent() {
```

Comment:
```suggestion
		} else if c.HasParent() {
```
?

Reply:
This check is correct: we need to update both affected subtrees. Suppose CQ moves from Cohort-A to Cohort-B. We need to remove resources/usage from Cohort-A, and add those to Cohort-B.